### ISSUE TITLE
nested folders don't work

### ISSUE THREAD

**Issue by BrannonKing** [dirk-thomas/vcstool#40]
I get the error below even with the `--force` option:
```
=== src/localization (hg) ===
Could not clone repository 'https://mercurial.asirobots.com:8443/scm/hg/localization': abort: destination '.' is not empty
```

It seems a little weird that it creates the "localization" folder prematurely. From my repos file:
```
  localization:
    type: hg
    url: https://mercurial.asirobots.com:8443/scm/hg/localization
    version: default
  localization/src/AutoGenFusor:
    type: tar
    url: https://builder.domain.local/guestAuth/repository/download/MatlabScripts_FusorCodeGen/1.5.5c5faf48e415.91/builderOutput.tgz
    version: 1.5.5c5faf48e415.91
  localization/src/AutoGenMatchMaker:
    type: tar
    url: https://builder.domain.local/guestAuth/repository/download/MatlabScripts_PFCodeGen/1.8.e8c4885e9249.103/builderOutput.tgz
    version: 1.8.e8c4885e9249.103
```

**Comment by dirk-thomas** [comment]
The problem seems to be not specific to `hg`. The same will be the case for e.g. `git`. E.g. replacing the `localization` entry with the following git repo has the same problem:

```
type: git
url: https://github.com/dirk-thomas/vcstool.git
version: master
```

I am not aware of a command line option to force `hg` or `git` to perform a `clone` in a not empty directory. Maybe the `--force` option can extended to delete the folder in that case (since the user explicitly asked for it) (lets call it feature A).

For your case that would still not work since the other entries will create content in a subfolder. As long as `vcs import` is being run multithreaded / with multiple parallel workers it would still fail. `vcs` would need to process the subfolder *after* the parent folder (lets call this feature B).

**Comment by BrannonKing** [comment]
I'm currently working around this like so:
1. Put my dependencies in folders that don't conflict.
2. Add symlinks (`ln -srd src/deps/AutoGen* -t src/localization/src/`) later in my overall build script.

**Comment by dirk-thomas** [comment]
I have created #41 to implement the feature identified as B (sequentially process nested entries). I have tried it for your example but only from scratch. Maybe you can give it a try if it works for you?

**Comment by dirk-thomas** [comment]
@BrannonKing It would be great it you could give #41 a try and comment if it satisfies your use case.

**Comment by dirk-thomas** [comment]
Addressed by #41. Released in 0.1.25.
